src/yz_app.erl

```diff
@@ -58,16 +60,46 @@ start(_StartType, _StartArgs) ->
             maybe_setup(Enabled),
 
             %% Now everything is started, permit usage by KV/query
-            yokozuna:enable(index),
-            yokozuna:enable(search),
+            enable_components(),
             {ok, Pid};
         Error ->
             Error
     end.
 
+%% @doc Prepare to stop - called before the supervisor tree is shutdown
+prep_stop(State) ->
+    try %% wrap with a try/catch - application carries on regardless,
+        %% no error message or logging about the failure otherwise.
+        lager:info("Stopping application yokozuna.\n", []),
+        ok = riak_api_pb_service:deregister(?QUERY_SERVICES),
+        ok = riak_api_pb_service:deregister(?ADMIN_SERVICES),
+        ok = yz_solrq_drain_mgr:drain(),
```

Comment:
Yeah, I considered removing all of the `ok = ` pattern matches but left them there in the end since they were in the original `stop` code. What do you think? Should I zap them all? Also, any thoughts on the order of operations in `prep_stop`?